app/jobs/calculate_all_metrics_job.rb

```diff
@@ -1,17 +1,23 @@
 # frozen_string_literal: true
 
-require "rake"
-
 class CalculateAllMetricsJob < ApplicationJob
-  queue_as :scheduled
+  sidekiq_options retry: 1, queue: :metrics
```

Comment:
I would use something like https://github.com/veeqo/activejob-uniqueness instead

Reply:
Here objective is to reduce retry count from 25 to 1 because each Job will be run every day and retry max 25 times if failed.

To me it is just Sidekiq configuration, not a uniqueness need. 
Maybe I don't get your point